src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -72,6 +72,14 @@ type commitLogSource struct {
 	newReaderFn     newReaderFn
 
 	metrics commitLogSourceMetrics
+
+	// Cache the results so the commit log is not read twice. This source is special since it reads the entire
+	// time series range, irrespective of the requested time range. This is required since the commit log may hold
+	// cold writes that have not been committed to a cold block. Another source might report it fulfilled a cold block,
+	// but it could be missing writes only available in the commit log.
+	//
+	// The bootstrapper is single threaded so we don't need a mutex for this.
+	bootstrapResults bootstrap.NamespaceResults
```

Comment:
What's the lifecycle of this? Any chance a downstream consumer will release the results in this map between te two reads?